File: SubscriptionApi/SubscriptionApi/Subscriptions/SubscriptionDataCorrectnessCriteria.cs
using SubscriptionApi.ResponseBuilders;

namespace SubscriptionApi.Subscriptions
{
  public class SubscriptionDataCorrectnessCriteria
  {
    public void ValidateSessionId(string subscriptionId, SubscriptionValidationResults results)
    {
      if (string.IsNullOrEmpty(subscriptionId))
      {
        results.NotValid("Subscription ID");
      }
    }

    public void ValidateUserName(string userName, SubscriptionValidationResults results)
    {
      if (string.IsNullOrEmpty(userName))
      {
        results.NotValid("user name");
      }
    }
  }
}

File: Java/factories/_01_no_factories/src/main/java/Subscriptions/SubscriptionDataCorrectnessCriteria.java
package subscriptions;

import responseBuilders.SubscriptionValidationResults;

public class SubscriptionDataCorrectnessCriteria implements DataCorrectnessCriteria {
    @Override
    public void validateSessionId(String subscriptionId, SubscriptionValidationResults results) {
        if ("".equals(subscriptionId) || subscriptionId == null) {
            results.notValid("Subscription ID");
        }
    }

    @Override
    public void validateUserName(String userName, SubscriptionValidationResults results) {
        if ("".equals(userName) || userName == null) {
            results.notValid("user name");
        }
    }
}
